data-prepper-plugins/elasticsearch/src/main/java/com/amazon/dataprepper/plugins/sink/elasticsearch/ElasticsearchSink.java

```diff
@@ -70,8 +71,12 @@ public ElasticsearchSink(final PluginSetting pluginSetting) {
     this.documentIdField = esSinkConfig.getIndexConfiguration().getDocumentIdField();
     try {
       start();
-    } catch (final IOException e) {
-      throw new RuntimeException(e.getMessage(), e);
+    } catch (final Exception e) {
+      try {
+        throw new RuntimeException(e.getMessage(), e);
+      } finally {
+        this.shutdown();
+      }
```

Comment:
Why change the exception type here?

Also can't we just add a `finally` block to the original try/catch, instead of introducing a new try/finally?

Reply:
> Why change the exception type here?

If you mean from IOException to Exception, this is to capture `ElasticsearchException` from `start()`.

> Also can't we just add a finally block to the original try/catch, instead of introducing a new try/finally?
The main reason is that finally will close the REST Client even if no exception is detected. Thus I have to do this. example:

```
try {
    System.out.println("a");
} finally {
    System.out.println("b");
}
```
both a and b will be printed.

Reply:
Ah, I misinterpreted `start()` to be a long-running operation, something to be interrupted. This is more like a initialization method. Potentially suggest renaming the method as such (doesn't have to be now).

How about just 
```
this.shutdown();
throw new RuntimeException(e.getMessage(), e);
```
to avoid the new try/finally then?

Reply:
Will do renaming.

> How about just
```
this.shutdown();
throw new RuntimeException(e.getMessage(), e);
```
to avoid the new try/finally then?

This is what I initially do. Then I realized any runtime exception from `shutdown` could overwrite the captured exception. But I probably overthought here.